subsys/bluetooth/controller/hci/hci.c

```diff
@@ -2916,7 +2916,12 @@ static void le_df_connection_iq_report(struct node_rx_pdu *node_rx, struct net_b
 
 	lll = iq_report->hdr.rx_ftr.param;
 
+#if defined(CONFIG_BT_PHY_UPDATE)
 	phy_rx = lll->phy_rx;
+#else
+	phy_rx = PHY_1M;
+#endif /* CONFIG_BT_PHY_UPDATE */
+
 	/* Make sure the report is generated for connection on PHY UNCODED */
 	LL_ASSERT(phy_rx != PHY_CODED);
```

Comment:
If PHY can only be 1M if `CONFIG_BT_PHY_UPDATE` is not enabled, then this assert can be moved up to the same guard.

Reply:
Actually I should have used `CONFIG_BT_CTLR_PHY` instead of `CONFIG_BT_PHY_UPDATE`. The latter one depends on the `CONFIG_BT_CTLR_PHY`. 

Yes, if the `CONFIG_BT_PHY_UPDATE` and `CONFIG_BT_CTLR_PHY` are not enabled the support for PHY2M and PHY CODED is disabled: https://github.com/zephyrproject-rtos/zephyr/blob/731241f8d049924591cf4f46dda26ddd0782d8c7/subsys/bluetooth/controller/Kconfig#L450-L462